### PR TITLE
Modal

### PR DESCRIPTION
**Pull request by nathanyoung**

### CODE REVIEW

File: src/components/Modal/Modal.stories.jsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import Modal from './Modal';
+import Button from '../Button/Button';
+
+export default {
+  title: 'Components/Modal/For Chromatic',
+  component: Modal,
+  parameters: {
+    chromatic: { delay: 1000 },
```
Comment (nathanyoung) [inline src/components/Modal/Modal.stories.jsx:9]: @juanfabrega this doesn't seem to be working. I tried following these instructions:

https://www.chromatic.com/docs/delay#delay-all-stories-of-a-component

**Comment by nathanyoung** [comment]
https://github.com/palmetto/palmetto-components/issues/293

File: src/components/Modal/Modal.module.scss [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+:global([data-reach-dialog-overlay]) {
+  display: flex;
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  align-items: flex-end;
+  z-index: var(--size-z-index-overlay);
+  background: hsla(0, 0%, 0%, 0.33);
```
Comment (juanfabrega) [inline src/components/Modal/Modal.module.scss:44]: Might be worth getting some tokens that represent semi-transparent colors. Not for this PR.

File: src/components/Modal/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import React, { FC, ReactNode, RefObject } from 'react';
```
Comment (juanfabrega) [inline src/components/Modal/Modal.tsx:1]: Feels like we should allow prop spreading down to the DialogOverlay to cover any options that are exposed by reach.

**Review by juanfabrega — APPROVED** [review]

**Pull request merged**